src/main/java/uk/gov/digital/ho/hocs/casework/api/CorrespondentService.java

```diff
@@ -60,6 +61,27 @@ void createCorrespondent(UUID caseUUID, String correspondentType, String fullnam
         log.info("Created Correspondent: {} for Case: {}", correspondent.getUuid(), caseUUID, value(EVENT, CORRESPONDENT_CREATED));
     }
 
+    void updateCorrespondent(UUID caseUUID, UUID correspondentUUID, UpdateCorrespondentRequest updateCorrespondentRequest){
+        log.debug("Updating Correspondent: {} for Case: {}", correspondentUUID, caseUUID);
+        Correspondent correspondent = getCorrespondent(caseUUID, correspondentUUID);
+        correspondent.setFullName(updateCorrespondentRequest.getFullname());
+        correspondent.setAddress1(updateCorrespondentRequest.getAddress1());
+        correspondent.setAddress2(updateCorrespondentRequest.getAddress2());
+        correspondent.setAddress3(updateCorrespondentRequest.getAddress3());
+        correspondent.setPostcode(updateCorrespondentRequest.getPostcode());
+        correspondent.setTelephone(updateCorrespondentRequest.getTelephone());
+        correspondent.setEmail(updateCorrespondentRequest.getEmail());
+        correspondent.setReference(updateCorrespondentRequest.getReference());
+
+        try {
+            correspondentRepository.save(correspondent);
+            auditClient.createCorrespondentAudit(correspondent);
```

Comment:
As discussed, this should be updated to auditClient.updateCorrespondentAudit(...). We'll also have to make corresponding changes in hocs-audit to make sure these events get fetched on Correspondent export.

Reply:
updated as discussed